--- configure	Tue Apr  1 04:48:24 2008
+++ configure.new	Wed Apr  9 16:42:42 2008
@@ -21759,10 +21759,10 @@
 -Wpointer-arith -Wwrite-strings -Wstrict-prototypes \
 -Wmissing-prototypes -Wmissing-declarations -Wnested-externs \
 -Wpacked -Wswitch-enum -Wmissing-format-attribute \
--Wstrict-aliasing=2 -Winit-self -Wunsafe-loop-optimizations \
+-Wstrict-aliasing=2 -Winit-self \
 -Wdeclaration-after-statement -Wold-style-definition \
--Wno-missing-field-initializers -Wno-unused-parameter \
--Wno-attributes -Wno-long-long -Winline -fno-strict-aliasing"
+-Wno-unused-parameter \
+-Wno-long-long -Winline -fno-strict-aliasing"
 
 # invalidate cached value if MAYBE_WARN has changed
 if test "x$enchant_cv_warn_maybe" != "x$CC_MAYBE_WARN"; then
@@ -21852,9 +21852,9 @@
 -Wsign-compare \
 -Wpointer-arith -Wwrite-strings \
 -Wpacked -Wswitch-enum -Wmissing-format-attribute \
--Wstrict-aliasing=2 -Winit-self -Wunsafe-loop-optimizations \
--Wno-missing-field-initializers -Wno-unused-parameter \
--Wno-attributes -Wno-long-long -Winline -fno-strict-aliasing"
+-Wstrict-aliasing=2 -Winit-self \
+-Wno-unused-parameter \
+-Wno-long-long -Winline -fno-strict-aliasing"
 
 # invalidate cached value if MAYBE_WARN has changed
 if test "x$enchant_cxx_cv_warn_maybe" != "x$CXX_MAYBE_WARN"; then
