--- src/enchant-1.1.6/src/enchant.c	Wed Mar  9 07:19:50 2005
+++ enchant-1.1.6/src/enchant.c	Wed Jun 29 14:43:52 2005
@@ -128,6 +128,26 @@
 	if (module_dir)
 		return module_dir;
 
+#ifdef _WIN32
+	/* Dynamically locate DLL and search for modules relative to it. */
+	char dll_path[MAX_PATH];
+	char *dll_dir;
+	HINSTANCE hInstance = GetModuleHandle("libenchant-1");
+	if(hInstance != NULL)
+		{
+		if(GetModuleFileName(hInstance,dll_path,MAX_PATH))
+			{
+			/* Create dir string and return */
+			dll_dir = g_path_get_dirname(dll_path);
+			module_dir = g_build_filename(dll_dir,"enchant",NULL);
+			g_free(dll_dir);
+			return module_dir;
+			}
+		}
+#endif
+
 #ifdef ENCHANT_GLOBAL_MODULE_DIR
 	return g_strdup (ENCHANT_GLOBAL_MODULE_DIR);
 #else
@@ -147,6 +167,10 @@
 	if (ordering_dir)
 		return ordering_dir;
 
+#ifdef _WIN32
+	/* TODO */
+#endif
+
 #ifdef ENCHANT_GLOBAL_ORDERING
 	return g_strdup (ENCHANT_GLOBAL_ORDERING);
 #else
@@ -200,7 +224,12 @@
 ENCHANT_MODULE_EXPORT (char *)
 enchant_get_registry_value (const char * const prefix, const char * const key)
 {
-	return enchant_get_registry_value_ex (0, prefix, key);
+	char *val;
+	val = enchant_get_registry_value_ex(1, prefix, key);
+	if(val == NULL) {
+		val = enchant_get_registry_value_ex (0, prefix, key);
+	}
+	return val;
 }
 
 /**
