--- enchant-1.1.5/src/enchant.c	Tue Nov 30 11:10:31 2004
+++ enchant-1.1.5.new/src/enchant.c	Wed Dec 29 05:34:25 2004
@@ -165,6 +165,7 @@
 	HKEY baseKey;
 	unsigned long lType;	
 	DWORD dwSize;
+	char* keyName;
 	BYTE* szValue = NULL;
 
 	if (current_user)
@@ -172,7 +173,8 @@
 	else
 		baseKey = HKEY_LOCAL_MACHINE;
 
-	if(RegOpenKeyEx(baseKey, "Software\\Enchant", 0, KEY_READ, &hKey) == ERROR_SUCCESS)
+	keyName = g_strdup_printf("Software\\Enchant\\%s", prefix);
+	if(RegOpenKeyEx(baseKey, keyName, 0, KEY_READ, &hKey) == ERROR_SUCCESS)
 		{
 			// Determine size of string
 			if(RegQueryValueEx( hKey, key, NULL, &lType, NULL, &dwSize) == ERROR_SUCCESS)
@@ -181,7 +183,7 @@
 					RegQueryValueEx(hKey, key, NULL, &lType, szValue, &dwSize);
 				}
 		}
-	
+	g_free(keyName);
 	return (char *)szValue;
 #endif
 }
