--- enchant-1.1.6/src/ispell/ispell_checker.cpp	Wed Feb  9 14:45:00 2005
+++ enchant-1.1.7/src/ispell/ispell_checker.cpp	Fri Jul  1 15:16:48 2005
@@ -363,10 +363,22 @@
 #ifndef XP_TARGET_COCOA
 	char * ispell_prefix = NULL;
 
+	/* Look for explicitly set registry values */
 	ispell_prefix = enchant_get_registry_value ("Ispell", "Data_Dir");
 	if (ispell_prefix)
 		return ispell_prefix;
 #endif
+
+	/* Dynamically locate library and search for modules relative to it. */
+	char * enchant_prefix = NULL;
+	enchant_prefix = enchant_get_prefix_dir();
+	if(enchant_prefix)
+		{
+			ispell_prefix = g_build_filename(enchant_prefix,"share","enchant","ispell",NULL);
+			g_free(enchant_prefix);
+			return ispell_prefix;
+		}
+
 #ifdef ENCHANT_ISPELL_DICT_DIR
 	return g_strdup (ENCHANT_ISPELL_DICT_DIR);
 #else
