--- Makefile.in.orig	Tue Sep 18 18:56:16 2007
+++ Makefile.in	Tue Sep 18 18:57:00 2007
@@ -59,7 +59,7 @@
 am__DEPENDENCIES_1 =
 libenchant_myspell_la_DEPENDENCIES = $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(top_builddir)/src/libenchant.la
-am_libenchant_myspell_la_OBJECTS = affentry.lo affixmgr.lo csutil.lo \
+am_libenchant_myspell_la_OBJECTS = affentry.lo affixmgr.lo phonet.lo csutil.lo \
 	dictmgr.lo hashmgr.lo hunspell.lo suggestmgr.lo \
 	myspell_checker.lo
 libenchant_myspell_la_OBJECTS = $(am_libenchant_myspell_la_OBJECTS)
@@ -241,6 +241,7 @@
 libenchant_myspell_la_SOURCES = \
 	affentry.hxx		\
 	affixmgr.hxx		\
+	phonet.hxx		\
 	atypes.hxx		\
 	baseaffix.hxx		\
 	csutil.hxx		\
@@ -252,6 +253,7 @@
 	suggestmgr.hxx		\
 	affentry.cxx		\
 	affixmgr.cxx		\
+	phonet.cxx		\
 	csutil.cxx		\
 	dictmgr.cxx		\
 	hashmgr.cxx		\
@@ -338,6 +340,7 @@
 
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/affentry.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/affixmgr.Plo@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/phonet.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/csutil.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/dictmgr.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/hashmgr.Plo@am__quote@
