--- enchant-1.6.0/src/myspell/myspell_checker.cpp	Fri Apr  2 07:53:37 2010
+++ enchant-patched/src/myspell/myspell_checker.cpp	Mon Aug 16 13:35:24 2010
@@ -410,6 +413,9 @@
 MySpellChecker::requestDictionary(const char *szLang)
 {
 	char *dic = NULL, *aff = NULL;
+	#if defined(_WIN32)
+	char *l_dic = NULL, *l_aff = NULL;
+	#endif
 
 	dic = myspell_request_dictionary (m_broker, szLang);
 	if (!dic)
@@ -420,7 +426,23 @@
 	strcpy(aff+len_dic-3, "aff");
 	if (g_file_test(aff, G_FILE_TEST_EXISTS))
 	{
+		#if defined(_WIN32)
+		l_dic = g_locale_from_utf8(dic,-1,NULL,NULL,NULL);
+		l_aff = g_locale_from_utf8(aff,-1,NULL,NULL,NULL);
+		if(l_dic && l_aff) {
+			myspell = new Hunspell(l_aff, l_dic);
+		} else {
+			myspell = new Hunspell(aff, dic);
+		}
+		if(l_dic) {
+			g_free(l_dic);
+		}
+		if(l_aff) {
+			g_free(l_aff);
+		}
+		#else
 		myspell = new Hunspell(aff, dic);
+		#endif
 	}
 	g_free(dic);
 	g_free(aff);
