--- src/enchant-1.1.6/src/myspell/myspell_checker.cpp	Thu Feb 10 13:04:20 2005
+++ enchant-1.1.6/src/myspell/myspell_checker.cpp	Thu Jun 30 06:31:24 2005
@@ -79,6 +79,23 @@
 	data_dir = enchant_get_registry_value ("Myspell", "Data_Dir");
 	if (data_dir)
 		return data_dir;
+#ifdef _WIN32
+	/* Dynamically locate DLL and search for modules relative to it. */
+	char dll_path[MAX_PATH];
+	char *dll_dir;
+	HINSTANCE hInstance = GetModuleHandle("libenchant_myspell-1");
+	if(hInstance != NULL)
+		{
+		if(GetModuleFileName(hInstance,dll_path,MAX_PATH))
+			{
+			/* Create dir string and return */
+			dll_dir = g_path_get_dirname(dll_path);
+			data_dir=g_build_filename(dll_dir,"..","myspell",NULL);
+			g_free(dll_dir);
+			return data_dir;
+			}
+		}
+#endif
 
 #ifdef ENCHANT_MYSPELL_DICT_DIR
 	return g_strdup (ENCHANT_MYSPELL_DICT_DIR);
