Index: src/enchant.c
===================================================================
--- src/enchant.c	2010-05-29 16:26:55.000000000 +1000
+++ enchant.c	2010-05-29 16:13:44.000000000 +1000
@@ -240,16 +240,20 @@
 
 #if defined(ENCHANT_GLOBAL_MODULE_DIR)
 	module_dirs = enchant_slist_append_unique_path (module_dirs, g_strdup (ENCHANT_GLOBAL_MODULE_DIR));
-#else
+#endif
+
 	/* Dynamically locate library and search for modules relative to it. */
 	prefix = enchant_get_prefix_dir();
 	if(prefix)
 		{
-			module_dir = g_build_filename(prefix,"lib","enchant",NULL);
+			if(sizeof(int*) == 4) { 
+				module_dir = g_build_filename(prefix,"lib","enchant",NULL);
+			} else {
+				module_dir = g_build_filename(prefix,"lib64","enchant",NULL);
+			}
 			g_free(prefix);
 			module_dirs = enchant_slist_append_unique_path (module_dirs, module_dir);
 		}
-#endif
 
 	return module_dirs;
 }
