--- src/myspell/myspell_checker.cpp	2005-06-30 16:58:47.000000000 +1000
+++ src/myspell/myspell_checker.cpp.new	2005-07-01 14:29:05.000000000 +1000
@@ -76,10 +76,22 @@
 {
 	char * data_dir = NULL;
 
+	/* Look for explicitly set registry values */
 	data_dir = enchant_get_registry_value ("Myspell", "Data_Dir");
 	if (data_dir)
 		return data_dir;
 
+	/* Dynamically locate library and search for modules relative to it. */
+	char * prefix = NULL;
+	prefix = enchant_get_prefix_dir();
+	if(prefix)
+		{
+			data_dir = g_build_filename(prefix,"share","enchant","myspell",NULL);
+			g_free(prefix);
+			return data_dir;
+		}
+
+
 #ifdef ENCHANT_MYSPELL_DICT_DIR
 	return g_strdup (ENCHANT_MYSPELL_DICT_DIR);
 #else
